Mathlib/Topology/UniformSpace/Cauchy.lean

```diff
@@ -36,6 +36,12 @@ def IsComplete (s : Set α) :=
   ∀ f, Cauchy f → f ≤ 𝓟 s → ∃ x ∈ s, f ≤ 𝓝 x
 #align is_complete IsComplete
 
+/-- A version of `lim` that deduces `Nonempty` from `Cauchy`.
+
+This is an abbreviation, so that we do not need to repeat all API lemmas. -/
+protected noncomputable abbrev Cauchy.lim (f : Filter α) (hf : Cauchy f) : α :=
```

Comment:
Why not make `f` implicit here? Apparently it doesn't change anything in practice, I don't really understand why though.

Also, what do you think about defining instead `Filter.lim'`, which takes a `NeBot f` instance instead of `Nonempty α` ?

Reply:
Ah sure I'm stupid. Then what we would want is probably my `lim'` with the instance and still another `Cauchy.lim` abbreviation on top of it. So since we would need the extra abbreviation in any case I'd say sticking with the current setup is fine.

Reply:
I loogled for `lim` and `Cauchy`. They're used together twice. So, I just added `haveI`s instead of an abbreviation.